pallets/settlement/src/lib.rs

```diff
@@ -540,85 +536,79 @@ decl_module! {
             valid_from: Option<T::Moment>,
             legs: Vec<Leg<T::Balance>>
         ) -> DispatchResult {
-            let sender = ensure_signed(origin)?;
-            let did = Context::current_identity_or::<Identity<T>>(&sender)?;
+            let did = Identity::<T>::ensure_origin_call_permissions(origin)?.primary_did;
             Self::base_add_instruction(did, venue_id, settlement_type, valid_from, legs)?;
             Ok(())
         }
 
-        /// Adds and authorizes a new instruction.
+        /// Adds and provide affirmation to a new instruction.
         ///
         /// # Arguments
         /// * `venue_id` - ID of the venue this instruction belongs to.
         /// * `settlement_type` - Defines if the instruction should be settled
-        ///    immediately after receiving all auths or waiting till a specific block.
+        ///    immediately after receiving all affirms or waiting till a specific block.
         /// * `valid_from` - Optional date from which people can interact with this instruction.
         /// * `legs` - Legs included in this instruction.
-        /// * `portfolios` - Portfolios that the sender controls and wants to authorize for this instruction
+        /// * `portfolios` - Portfolios that the sender controls and wants to affirm this instruction
```

Comment:
This doesn't sound right. Maybe try something like
```suggestion
        /// * `portfolios` - Portfolios that the sender controls and wants to use in this affirmation
```